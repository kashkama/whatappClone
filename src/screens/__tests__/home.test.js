import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Home from "./../home";

configure({ adapter: new Adapter()})

it("renders home component correctly", () => {
    const tree = renderer.create(
        <Home/>
    );
    expect(tree).toMatchSnapshot();
});

it("has proper navigation options", () => {
    expect(Home.navigationOptions).toMatchSnapshot();
})

it("navigate on press", () => {
    const navigate = jest.fn();
    const homeScreen = shallow(
        <Home navigation={{ navigate }}/>
    );
    homeScreen.find("Button").simulate("press");
    expect(navigate).toBeCalledWith("Chat", {name: "john"})
})