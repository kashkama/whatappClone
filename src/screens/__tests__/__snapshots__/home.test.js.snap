// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has proper navigation options 1`] = `
Object {
  "title": "Home Screen",
}
`;

exports[`renders home component correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          john
        </Text>
      </View>
    </View>
  </View>
</View>
`;
